File: Searching/Linear Search/LinearSearch.py
def linear_search(arr, target):
    """
    Perform linear search to find the target element in the array.

    Args:
    arr (list): List of elements to search through.
    target (any): Element to search for.

    Returns:
    int: Index of the target element if found, else -1.
    """
    # Loop through each element in the array
    for index in range(len(arr)):
        # Check if the current element matches the target
        if arr[index] == target:
            return index  # Return the index if target is found

    return -1  # Return -1 if target is not found

# Example usage of Linear Search
arr = [10, 20, 30, 40, 50]
target = 30
result = linear_search(arr, target)
if result != -1:
    print(f"Element {target} found at index {result}")
else:
    print(f"Element {target} not found in the array")

File: Searching/Linear Search/LinearSearch.cpp
#include<iostream>
using namespace std;

void LinearSearch(int arr[], int len, int item)
{
    for (int i = 0; i < len; i++)
    {
        if(arr[i] == item)
        {
            cout <<  item  << "Found at Index: " << i;
            return;
        }
    }
    cout << "Not Found";
}
int main()
{
    int arr[] = {10, 5, 15, 21, -6, 7};
    //calculating length of array
    int len = sizeof(arr)/sizeof(arr[0]);

    //item to be searched
    int item = 21;
    LinearSearch(arr, len, item);

    return 0;
}
//Space complexity : O(N)
//Time Complexity : O(N)